### PR TITLE
Handle ContainerNotFound exception in ObjectResource HEAD

### PR DESCRIPTION
**Pull request by timuralp**
When processing HEAD for a blob, we should handle the ContainerNotFound exception.

### CODE REVIEW

**Comment by kahing** [comment]
:+1:

File: src/test/java/com/bouncestorage/swiftproxy/v1/ObjectResourceTest.java [hunk 1]
```diff
@@ -76,6 +76,24 @@ public void testPut() throws Exception {
...
+    @Test
+    public void testMissingContainerHEAD() throws Exception {
+        Response resp = target.path(Joiner.on("/").join(TestUtils.ACCOUNT_PATH, "foo", "bar"))
+                .request()
+                .header("x-auth-token", authToken)
+                .head();
+        assertThat(resp.getStatus()).isEqualTo(Response.Status.NOT_FOUND.getStatusCode());
+    }
+
+    @Test
```
Comment (andrewgaul) [inline src/test/java/com/bouncestorage/swiftproxy/v1/ObjectResourceTest.java]: These kinds of conformance tests are best added in the Swift integration tests -- the unit tests should just do basic sanity checks like in S3Proxy.  Does the former have coverage for this?

Comment (timuralp) [inline src/test/java/com/bouncestorage/swiftproxy/v1/ObjectResourceTest.java]: I will look into that. However, currently, SwiftProxy returns 500 -- Internal Server Error -- if this happens, which is wrong regardless of what Swift specifies.

Comment (timuralp) [inline src/test/java/com/bouncestorage/swiftproxy/v1/ObjectResourceTest.java]: Swift tests do have a test for this. Looks like this only comes up with jclouds transient and file system blob stores (https://issues.apache.org/jira/browse/JCLOUDS-193). I will remove the test from the PR.

**Comment by timuralp** [comment]
Rewrote to use the Jersey ExceptionMapper.

**Pull request merged**